mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/events/WorkerMetricsCollector.java

```diff
@@ -120,7 +120,11 @@ public void process(WorkerStatusEvent workerStatusEvent) {
             final WorkerId workerId = workerStatusEvent.getWorkerId();
             Preconditions.checkNotNull(jobWorkers.get(jobId));
             final IMantisWorkerMetadata metadata = jobWorkers.get(jobId).get(workerId);
-            Preconditions.checkNotNull(metadata);
+            if (metadata == null) {
+                log.warn("Unknown workerId: {} for metrics collector in job: {}", workerId, jobId);
```

Comment:
In what situation will this be null? Can you raise it to the error level?

Reply:
When some workers are in-transition and being replace this can happen. And npe gets wired with radar auto alerts (email notification) when this happens.

Reply:
There are other logs in control plane to track those cases already iirc.